### PR TITLE
Feature: Additional Phone Highlight Implementation

### PR DESCRIPTION
**Pull request by ystanislavyk**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ before_install:
...
-  - docker run --rm travis-build /bin/bash -c "mkdir build-clang-debug && cd build-clang-debug && CXX=clang++-7 CC=clang-7 cmake -DCMAKE_BUILD_TYPE=Debug .. && make -j$(nproc) && ctest --output-on-failure && ./benchmark"
-  - docker run --rm travis-build /bin/bash -c "mkdir build-clang-release && cd build-clang-release && CXX=clang++-7 CC=clang-7 cmake -DCMAKE_BUILD_TYPE=Release .. && make -j$(nproc) && ctest --output-on-failure && ./benchmark"
-  - docker run --rm travis-build /bin/bash -c "mkdir build-gcc-debug && cd build-gcc-debug && CXX=g++-8 CC=gcc-8 cmake -DCMAKE_BUILD_TYPE=Debug .. && make -j$(nproc) && ctest --output-on-failure && ./benchmark"
-  - docker run --rm travis-build /bin/bash -c "mkdir build-gcc-release && cd build-gcc-release && CXX=g++-8 CC=gcc-8 cmake -DCMAKE_BUILD_TYPE=Release .. && make -j$(nproc) && ctest --output-on-failure && ./benchmark"
\ No newline at end of file
+  - docker run --rm travis-build /bin/bash -c "export NINJA_STATUS='[%f/%t] (%e)' && mkdir build-clang-tidy && cd build-clang-tidy && CXX=clang++-7 CC=clang-7 cmake -GNinja .. && ninja clang-tidy -j $(nproc) -v"
```
Comment (malirod) [inline .travis.yml]: @Morheit , 
-j is redundant with ninja,
-v - do we really need it?
if ninja status shown incremental build time then remove this. this is not what we need.

File: src/PhoneHighlightYS.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+{
+    if (phoneNumber.empty() || searchString.empty())
+        return HighlightRanges();
+
+    static std::u32string const ignoredChars = U" ()/#.\u00A0";
+
+    HighlightRanges highlightRanges;
+    std::pair<uint32_t, uint32_t> range;
+
+    uint8_t searchStringPos = 0;
```
Comment (malirod) [inline src/PhoneHighlightYS.cpp]: @Morheit , with uint8_t you assume that phone length is no longer than 255. But in fact we can feed phone with any size.

File: src/PhoneHighlightYS.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        if (IsSymbolIgnored(phoneNumber[phoneNumberPos], ignoredChars)) {
+            if (isMatchingStarted) {
+                CollectRange(args);
+            }
+            continue;
+        }
+
+        const bool matchingFound = (phoneNumber[phoneNumberPos] == searchString[searchStringPos]);
```
Comment (malirod) [inline src/PhoneHighlightYS.cpp]: @Morheit , please follow east const

File: src/PhoneHighlightYS.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+constexpr uint8_t kHighlightRanges = 5u;
+
+template <typename Args>
+void CollectRange(Args const& args)
+{
+    std::get<kMatchingStatus>(args) = false;
+    std::get<kHighlightRanges>(args).push_back(std::get<kRange>(args));
+}
+
+template <typename Symbol, typename IgnoredSymbols>
```
Comment (malirod) [inline src/PhoneHighlightYS.cpp]: @Morheit , why you need template?

File: src/PhoneHighlightYS.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        return HighlightRanges();
+
+    static std::u32string const ignoredChars = U" ()/#.\u00A0";
+
+    HighlightRanges highlightRanges;
+    std::pair<uint32_t, uint32_t> range;
+
+    uint8_t searchStringPos = 0;
+    bool isMatchingStarted = false;
+    auto args = std::tie(phoneNumber, searchString, searchStringPos, isMatchingStarted, range, highlightRanges);
```
Comment (malirod) [inline src/PhoneHighlightYS.cpp]: @Morheit , why not just create special struct and not bother with tie\tuple\template?

**Review by malirod — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (ystanislavyk) [reply .travis.yml]: @malirod, `-j`, `-v` and  `export NINJA_STATUS=...` were removed

File: src/PhoneHighlightYS.cpp (hunk 2 above)
Comment (ystanislavyk) [reply src/PhoneHighlightYS.cpp]: @malirod, `uint8_t` has been replaced by `uint32_t`

File: src/PhoneHighlightYS.cpp (hunk 3 above)
Comment (ystanislavyk) [reply src/PhoneHighlightYS.cpp]: @malirod, done

File: src/PhoneHighlightYS.cpp (hunk 4 above)
Comment (ystanislavyk) [reply src/PhoneHighlightYS.cpp]: @malirod, it was handy during the implementation. All templates have been removed

File: src/PhoneHighlightYS.cpp (hunk 5 above)
Comment (ystanislavyk) [reply src/PhoneHighlightYS.cpp]: @malirod, using of `std::tie` was replaced by special struct

**Pull request merged**